substrate/bin/utils/chain-spec-builder/bin/main.rs

```diff
@@ -71,6 +81,36 @@ fn main() -> Result<(), String> {
 			let _ = serde_json::from_str::<serde_json::Value>(&chain_spec.as_json(true)?)
 				.map_err(|e| format!("Conversion to json failed: {e}"))?;
 		},
+		ChainSpecBuilderCmd::ListPresets(ListPresetsCmd { runtime_wasm_path }) => {
+			let code = fs::read(runtime_wasm_path.as_path())
+				.map_err(|e| format!("wasm blob shall be readable {e}"))?;
+			let caller: GenesisConfigBuilderRuntimeCaller =
+				GenesisConfigBuilderRuntimeCaller::new(&code[..]);
+			let presets = caller
+				.preset_names()
+				.map_err(|e| format!("getting default config from runtime should work: {e}"))?;
+			let presets: Vec<String> = presets
+				.into_iter()
+				.map(|preset| {
+					String::from(
+						TryInto::<&str>::try_into(&preset)
+							.unwrap_or_else(|_| "cannot display preset id")
+							.to_string(),
+					)
+				})
+				.collect();
+			println!("Known presets are:\n{presets:#?}");
```

Comment:
Tiny nit: I was playing a bit with the chain-spec-builder, and thought about the output here. IMO we could have this as json output as it is to have it machine-readable (e.g. you can forward to a file and read it in another script). But then we should probably remove the "Known presets" part.

Or have it for manual use, but I think then it would be preferable to just have a plain list without anything, but really no strong opinion.

Reply:
done in: 23067cdc4fe165d4686f527f1aff63e4e665eb96